### PR TITLE
Open_Source_Indicator_Enrichment_with_Teams

### PR DESCRIPTION
**Pull request by mhofert-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 22.591ms

Post your results with screenshots in this PR using the following style:

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow1

#### Teams

teams1
teams2
teams3
teams4

#### Artifact

artifact1
artifact2
artifact3

### CODE REVIEW

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [alerting_and_notifications, threat_detection_and_response]
+  keywords: [enrichment, teams, url, ip, hash]
+  features: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Open_Source_Indicator_Enrichment_With_Slack
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml:15]:
```suggestion
  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Open_Source_Indicator_Enrichment_With_Teams
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon [hunk 2]
```diff
@@ -0,0 +1,2104 @@
+{
+  "kom": {
+    "komandVersion": "1.49.0-0-gc013d",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-04-20T14:55:53.986661133Z",
+    "workflowVersions": [
+      {
+        "name": "Open_Source_Indicator_Enrichment_with_Teams",
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon:8]:
```suggestion
        "name": "Open Source Indicator Enrichment with Teams",
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon [hunk 3]
```diff
@@ -0,0 +1,2104 @@
...
+            "nodeId": "00c2a382-37e6-44a4-a00f-5f31fc288f0e",
+            "name": "Lookup IP Output",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "analysis for:{{[bb63cb06-20d4-4166-b3c1-3abdd048da68].[argument]}}\n\n*WhoIs Results*\n\nOrg Details\nReg Date: {{[55db8044-0992-415e-b561-bcd83c33a12b].[regdate]}}\nOrg Name: {{[55db8044-0992-415e-b561-bcd83c33a12b].[orgname]}}\nNet Name: {{[55db8044-0992-415e-b561-bcd83c33a12b].[netname]}}\nNet Type: {{[55db8044-0992-415e-b561-bcd83c33a12b].[nettype]}}\nNetRange: {{[55db8044-0992-415e-b561-bcd83c33a12b].[netrange]}}\nLocation Details\nAddress: {{[55db8044-0992-415e-b561-bcd83c33a12b].[address]}}\nCity: {{[55db8044-0992-415e-b561-bcd83c33a12b].[city]}}\nCountry: {{[55db8044-0992-415e-b561-bcd83c33a12b].[country]}}\n\n*Dig Results*\n\nQuestion Asked: {{[d005440d-a757-43a2-8e32-0688fccc84d8].[question]}}\nAnswer Received: {{[d005440d-a757-43a2-8e32-0688fccc84d8].[answer]}}\nQuery Status: {{[d005440d-a757-43a2-8e32-0688fccc84d8].[status]}}\n"
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon:251]:
```suggestion
                "content": "Analysis for: {{[bb63cb06-20d4-4166-b3c1-3abdd048da68].[argument]}}\n\n*WhoIs Results*\n\nOrg Details\nReg Date: {{[55db8044-0992-415e-b561-bcd83c33a12b].[regdate]}}\nOrg Name: {{[55db8044-0992-415e-b561-bcd83c33a12b].[orgname]}}\nNet Name: {{[55db8044-0992-415e-b561-bcd83c33a12b].[netname]}}\nNet Type: {{[55db8044-0992-415e-b561-bcd83c33a12b].[nettype]}}\nNetRange: {{[55db8044-0992-415e-b561-bcd83c33a12b].[netrange]}}\nLocation Details\nAddress: {{[55db8044-0992-415e-b561-bcd83c33a12b].[address]}}\nCity: {{[55db8044-0992-415e-b561-bcd83c33a12b].[city]}}\nCountry: {{[55db8044-0992-415e-b561-bcd83c33a12b].[country]}}\n\n*Dig Results*\n\nQuestion Asked: {{[d005440d-a757-43a2-8e32-0688fccc84d8].[question]}}\nAnswer Received: {{[d005440d-a757-43a2-8e32-0688fccc84d8].[answer]}}\nQuery Status: {{[d005440d-a757-43a2-8e32-0688fccc84d8].[status]}}\n"
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon [hunk 4]
```diff
@@ -0,0 +1,2104 @@
...
+          },
+          "7b7289f0-68ba-4e18-aff1-d28798ec24ed": {
+            "nodeId": "7b7289f0-68ba-4e18-aff1-d28798ec24ed",
+            "name": "Lookup URL Output",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "analysis for:  {{[bb63cb06-20d4-4166-b3c1-3abdd048da68].[argument]}}\n\n*WhoIs Results*\nDomain Name: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[name]}}\nDomain Registrar: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[registrar]}}\nDomain Creation Date: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[creation_date]}}\nDomain Last Updated: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[last_updated]}}\nRegistrar Abuse Contact: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[registrar_abuse_contact_email]}}"
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon:1116]:
```suggestion
                "content": "Analysis for: {{[bb63cb06-20d4-4166-b3c1-3abdd048da68].[argument]}}\n\n*WhoIs Results*\nDomain Name: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[name]}}\nDomain Registrar: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[registrar]}}\nDomain Creation Date: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[creation_date]}}\nDomain Last Updated: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[last_updated]}}\nRegistrar Abuse Contact: {{[1ecb5f27-5497-4bc6-96dc-d25e40830738].[registrar_abuse_contact_email]}}"
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon [hunk 5]
```diff
@@ -0,0 +1,2104 @@
...
+              "slugVersion": "2.0.2",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.0.2/icon.png"
+            },
+            "identifier": "send_html_message",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "channel_name": "change_me",
+                "message_content": "This workflow will provide enrichment on URL's IP's and hashs\nCommands should be in the following format:<br>\n<br>\n!investigate \"command\" \"data\" <br>\n<br>\nexamples:<br>\n!investigate url rapid7.com<br>\n<br>\n!investigate ip 198.51.100.100<br>\n<br>\n!investigate hash 009f1e9b72cfb6daa3de82093a755bdb3685e0eb",
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/Open_Source_Indicator_Enrichment_with_Teams.icon:1213]:
```suggestion
                "message_content": "This workflow will provide enrichment on URLs, IP addresses, and hashes\nCommands should be in the following format:<br>\n<br>\n!investigate \"command\" \"data\" <br>\n<br>\nexamples:<br>\n!investigate url rapid7.com<br>\n<br>\n!investigate ip 198.51.100.100<br>\n<br>\n!investigate hash 009f1e9b72cfb6daa3de82093a755bdb3685e0eb",
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# Description
+
+Look up and enrich URLs, IP addresses, and hashes directly from Teams using open source threat intelligence tools
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md:3]:
```suggestion
Look up and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+such as Dig, DNS and Whois.
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md:4]:
```suggestion
such as Dig, DNS, and Whois.
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+# Key Features
+
+* IP address, URL, and hash enrichment from Teams
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md:8]:
```suggestion
* IP address, URL, and hash enrichment from Microsoft Teams
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  - name: workflow1.png
+    title: Workflow View
+  - name: teams1.png
+    title: Example Teams Output for URL
+  - name: teams2.png
+    title: Example Teams Output for Hash Enrichment
+  - name: teams3.png
+    title: Example Teams Output for IP Enrichment
+  - name: teams4.png
+    title: Example Teams Output for Help Enrichment
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml:33]:
```suggestion
    title: Example Teams Output for Help
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  - name: artifact1.png
+    title: Example Artifact for URL Enrichment
+  - name: artifact2.png
+    title: Example Artifact for Hash Enrichment
+  - name: artifact3.png
+    title: Example Artifact for IP Enrichment
+  - name: workflow1.png
+    title: Workflow View
+  - name: teams1.png
+    title: Example Teams Output for URL
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml:27]:
```suggestion
    title: Example Teams Output for URL Enrichment
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
+
+# Documentation
+
+## Setup
+
+Once the workflow has been downloaded, login to InsightConnect and "Import" it into the workflow builder.
+After import, you will initially be prompted to configure the connection for Teams.
+You will then need to configure the Teams trigger and actions.
+In the Team and chanel inputs replace `change_me` with the appropriate team and chanel.
```
Comment (tylerslijboom-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md:21]:
```suggestion
In the Team and chanel inputs replace `change_me` with the appropriate team and channel.
```

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+To run the workflow, in the configured channel  enter the command `!investigate <command> <data>`.
```
Comment (tylerslijboom-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md:23]:
```suggestion
To run the workflow, in the configured channel mention the slackbot and enter the command `!investigate <command> <data>`.
```

Comment (mhofert-r7) [reply workflows/Open_Source_Indicator_Enrichment_with_Teams/help.md]: @ bot style commands are not supported in teams. You must select a channel and then you just use !investigate <command> <data>

File: workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+name: Open_Source_Indicator_Enrichment_with_Teams
+title: "Open Source Indicator Enrichment with Teams"
+description: "Look up and enrich URLs, IP addresses, and hashes directly from Teams using open source threat intelligence tools such as Dig DNS and Whois."
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [alerting_and_notifications, threat_detection_and_response]
+  keywords: [enrichment, teams, url, ip, hash]
```
Comment (jschipp-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [enrichment, teams, work_from_home, phishing]
```

File: workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/Open_Source_Indicator_Enrichment_with_Microsoft_Teams.icon [hunk 14]
```diff
@@ -2,13 +2,13 @@
...
-    "exportedAt": "2020-04-20T14:55:53.986661133Z",
+    "exportedAt": "2020-04-21T15:53:54.301808878Z",
     "workflowVersions": [
       {
-        "name": "Open_Source_Indicator_Enrichment_with_Teams",
+        "name": "Open Source Indicator Enrichment with Microsoft Teams",
         "type": "runnable",
         "version": "",
-        "description": "",
+        "description": "Look up and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools such as Dig DNS and Whois.",
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/Open_Source_Indicator_Enrichment_with_Microsoft_Teams.icon:11]:
```suggestion
        "description": "Look up and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools such as Dig, DNS, and Whois.",
```

File: workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/help.md [hunk 15]
```diff
@@ -16,11 +16,11 @@ such as Dig, DNS and Whois.
...
-After import, you will initially be prompted to configure the connection for Teams.
-You will then need to configure the Teams trigger and actions.
-In the Team and chanel inputs replace `change_me` with the appropriate team and chanel.
+After import, you will initially be prompted to configure the connection for Microsoft Teams.
+You will then need to configure the Microsoft Teams trigger and actions.
+In the Team and chanel inputs replace `change_me` with the appropriate team and channel.
 
-To run the workflow, in the configured channel  enter the command `!investigate <command> <data>`.
+To run the workflow, in the configured channel enter the command `!investigate <command> <data>`.
  Example: `!investigate ip 8.8.8.8`
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/help.md]: Replace those lines with the following standardized messaging:
```
Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.

Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input with the preset text of `change_me`).

To run the workflow, send the command `!investigate <command> <data>` in your configured team and channel. The workflow will post after it has completed.
```

File: workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/Open_Source_Indicator_Enrichment_with_Microsoft_Teams.icon [hunk 16]
```diff
@@ -0,0 +1,2374 @@
...
+  "kom": {
+    "komandVersion": "1.49.1-0-g76758",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-04-21T21:13:57.990058603Z",
+    "workflowVersions": [
+      {
+        "name": "Open Source Indicator Enrichment with Microsoft Teams",
+        "type": "runnable",
+        "version": "",
+        "description": "Look up and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools such as Dig DNS and Whois.",
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/Open_Source_Indicator_Enrichment_with_Microsoft_Teams.icon:11]:
```suggestion
        "description": "Lookup and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools such as Dig, DNS, and Whois.",
```

File: workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/help.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
+# Description
+
+Look up and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/help.md:3]:
```suggestion
Lookup and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools
```

File: workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/help.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+|Plugin|Version|Count|
+|----|----|--------|
+|Dig|1.0.3|1|
+|Team Cymru MHR|1.0.3|1|
+|Microsoft Teams|2.0.2|6|
+|WHOIS|1.0.7|2|
+
+## Troubleshooting
+
+In some instances using copy & paste for `!investigate` commands many introduce unicode characters.
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/help.md:41]:
```suggestion
In some instances using copy & paste for `!investigate` commands may introduce unicode characters.
```

File: workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/workflow.spec.yaml [hunk 19]
```diff
@@ -0,0 +1,33 @@
+extension: workflow
+products: ["insightconnect"]
+name: Open_Source_Indicator_Enrichment_with_Microsoft_Teams
+title: "Open Source Indicator Enrichment with Microsoft Teams"
+description: "Look up and enrich URLs, IP addresses, and hashes directly from Teams using open source threat intelligence tools such as Dig DNS and Whois."
```
Comment (mrinehart-r7) [inline workflows/Open_Source_Indicator_Enrichment_with_Microsoft_Teams/workflow.spec.yaml:5]:
```suggestion
description: "Look up and enrich URLs, IP addresses, and hashes directly from Microsoft Teams using open source threat intelligence tools such as Dig, DNS, and Whois."
```

**Pull request merged**